PythonVisualizations/Stack.py

```diff
@@ -0,0 +1,322 @@
+
+    def cellCenter(self, index):  # Center point for array cell at index
+        half_cell_x = (self.CELL_WIDTH - self.CELL_BORDER) // 2
+        half_cell_y = (self.CELL_HEIGHT - self.CELL_BORDER) // 2
+
+        return add_vector(self.cellCoords(index), (half_cell_x, half_cell_y))
+
+    def createArrayCell(self, index):  # Create a box representing an array cell
+        cell_coords = self.cellCoords(index)
+        #print("cell_Coords", cell_coords, flush=True)
+        half_border = self.CELL_BORDER // 2
+        arrayCoords = add_vector(cell_coords,
+                                 (-half_border, -half_border,
+                                  self.CELL_BORDER - half_border, self.CELL_BORDER - half_border + 2))
+        #print("array Coords", arrayCoords, flush=True)
+        rect = self.canvas.create_rectangle(arrayCoords,
+                                            fill=None, outline=self.CELL_BORDER_COLOR, width=self.CELL_BORDER)
+        self.canvas.lower(rect)
+
+        return rect
+
+    def createCellValue(self, indexOrCoords, key, color=None):
+        """Create new canvas items to represent a cell value.  A square
+        is created filled with a particular color with an text key centered
+        inside.  The position of the cell can either be an integer index in
+        the Array or the bounding box coordinates of the square.  If color
+        is not supplied, the next color in the palette is used.
+        An event handler is set up to update the VisualizationApp argument
+        with the cell's value if clicked with any button.
+        Returns the tuple, (square, text), of canvas items
+        """
+        # Determine position and color of cell
+        if isinstance(indexOrCoords, int):
+            rectPos = self.cellCoords(indexOrCoords)
+            valPos = self.cellCenter(indexOrCoords)
+        else:
+            rectPos = indexOrCoords
+            valPos = divide_vector(add_vector(rectPos[:2], rectPos[2:]), 2)
+
+        if color is None:
+            # Take the next color from the palette
+            color = drawable.palette[Stack.nextColor]
+            Stack.nextColor = (Stack.nextColor + 1) % len(drawable.palette)
+        cell_rect = self.canvas.create_rectangle(
+            *rectPos, fill=color, outline='', width=0)
+        cell_val = self.canvas.create_text(
+            *valPos, text=str(key), font=self.VALUE_FONT, fill=self.VALUE_COLOR)
+        handler = lambda e: self.setArgument(str(key))
+        for item in (cell_rect, cell_val):
+            self.canvas.tag_bind(item, '<Button>', handler)
+
+        return cell_rect, cell_val
+
+    def display(self):
+        self.canvas.delete("all")
+
+        for i in range(self.size):  # Draw grid of cells
+            self.createArrayCell(i)
+
+        # go through each Drawable in the list
+        for i, n in enumerate(self.list):
+            print("this is i and n", i, n)
```

Comment:
Print statements for debugging purposes should be removed from the code (or at least commented out) before committing the files to a branch that will be merged into `master`.